charts/istio/istiod/templates/deployment-master.yaml

```diff
@@ -0,0 +1,258 @@
+# This Deployment runs only on the master nodes
+# It is the same as the other Deployment except for a hardcoded nodeAffinity and different labels
+# These Istiod pods will be used by the kube-apiserver for sidecar injection and nothing else
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: istiod{{- if not (eq .Values.revision "") }}-{{ .Values.revision }}{{- end }}-master
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: istiod-master
+    istio.io/rev: {{ .Values.revision | default "default" }}
+    install.operator.istio.io/owning-resource: {{ .Values.ownerName | default "unknown" }}
+    operator.istio.io/component: "Pilot"
+    istio: pilot-master
+    release: {{ .Release.Name }}
+{{- range $key, $val := .Values.pilot.deploymentLabels }}
+    {{ $key }}: "{{ $val }}"
+{{- end }}
+spec:
+{{- if not .Values.pilot.autoscaleEnabled }}
+{{- if .Values.pilot.replicaCount }}
+  replicas: {{ .Values.pilot.replicaCount }}
+{{- end }}
+{{- end }}
+  strategy:
+    rollingUpdate:
+      maxSurge: {{ .Values.pilot.rollingMaxSurge }}
+      maxUnavailable: {{ .Values.pilot.rollingMaxUnavailable }}
+  selector:
+    matchLabels:
+      {{- if ne .Values.revision "" }}
+      app: istiod-master
+      istio.io/rev: {{ .Values.revision | default "default" }}
+      {{- else }}
+      istio: pilot-master
+      {{- end }}
+  template:
+    metadata:
+      labels:
+        app: istiod-master
+        istio.io/rev: {{ .Values.revision | default "default" }}
+        install.operator.istio.io/owning-resource: {{ .Values.ownerName | default "unknown" }}
+        sidecar.istio.io/inject: "false"
+        operator.istio.io/component: "Pilot"
+        {{- if ne .Values.revision "" }}
+        istio: istiod-master
+        {{- else }}
+        istio: pilot-master
+        {{- end }}
+        {{- range $key, $val := .Values.pilot.podLabels }}
+        {{ $key }}: "{{ $val }}"
+        {{- end }}
+      annotations:
+        {{- if .Values.meshConfig.enablePrometheusMerge }}
+        prometheus.io/port: "15014"
+        prometheus.io/scrape: "true"
+        {{- end }}
+        sidecar.istio.io/inject: "false"
+        {{- if .Values.pilot.podAnnotations }}
+{{ toYaml .Values.pilot.podAnnotations | indent 8 }}
+        {{- end }}
+    spec:
+      affinity:
+        # ensure that this deployment only runs on master nodes
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: node-role.kubernetes.io/control-plane
+                operator: In
+                values:
+                - "true"
```

Comment:
I think there would also need to be a toleration for this to get scheduled properly.

Reply:
The tolerations are specified in the values:
https://github.com/acorn-io/hub-system-charts/pull/20/files/f0304be92c0b78b9f02fee479e31dc7411688867#diff-d59d6c775c45e7acc1f8d1c229db0b547b32b1f05f643f64c1404309dcb40b20